api/routes/recipesRouter.js

```diff
@@ -0,0 +1,121 @@
+const express = require('express')
+
+const Recipes = require('../models/recipesModel.js')
+
+const router = express.Router()
+
+router.use(express.json())
+
+router.get('/', async (req, res) => {
+    try {
+        const recipes = await Recipes.find()
+        res.status(200).json(recipes)
+    } catch (error) {
+        console.log(error)
+    }
+})
+
+router.get('/:id',  validateRecipeId, async (req, res) => {
+    try {
+        const recipe = await Recipes.findById(req.recipe.id)
+        res.status(200).json(recipe)
+    } catch (error) {
+        console.log(error)
+        res.status(500).json(error)
+    }
+})
+
+router.get('/:id/ingredients',  validateRecipeId, async (req, res) => {
+    try {
+        const ingredientsList = await Recipes.findIngredientsById(req.recipe.id)
+        const dish = await Recipes.findDishesById(req.recipe.id)
+        const recipe = await Recipes.findById(req.recipe.id)
+        const shoppingList = {
+            dish_name: dish.name,
+            recipe_name: recipe.name,

```

Comment:
This all works great, see if you like the pattern I suggested above in the model files to accomplish the same thing